File: aspnet-in-practice/14/VB/Cache/Cache/Providers/InMemory/MemoryCacheBuilder.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Runtime.Caching

Namespace Providers.InMemory
	Public Class MemoryCacheBuilder
		Implements ICacheBuilder
		Public Sub New()
		End Sub

		Public Function GetInstance() As ObjectCache Implements ICacheBuilder.GetInstance
			Return MemoryCache.Default
		End Function

		Public ReadOnly Property DefaultRegionName() As String Implements ICacheBuilder.DefaultRegionName
			Get
				Throw New NotSupportedException()
			End Get
		End Property
	End Class
End Namespace

File: aspnet-in-practice/14/CS/Cache/Cache/Cache/InMemory/MemoryCacheBuilder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Runtime.Caching;

namespace Cache.Providers.InMemory
{
	public class MemoryCacheBuilder : ICacheBuilder
	{
		public MemoryCacheBuilder() { }

		public ObjectCache GetInstance()
		{
			return MemoryCache.Default;
		}

		public string DefaultRegionName
		{
			get { throw new NotSupportedException(); }
		}
	}
}